wgsl/media-type-registration.txt

```diff
@@ -0,0 +1,65 @@
+Type name: text
+
+Subtype name: wgsl
+
+Required parameters: N/A
+
+Optional parameters: None
+
+Encoding considerations: binary
+
+  WGSL is Unicode text using the UTF-8 encoding. See [WGSL] Section
+  3. Textual Structure.
+
+Security considerations:
+
+  WebGPU Shading Language (WGSL) is a programming language for GPU
+  code to be executed in the context of the WebGPU API. For security
+  considerations, see [WebGPU] Section 2.1 Security Considerations.
+
+Interoperability considerations:
+ 
+  Implementations of WebGPU may have different capabilities, and
+  these differences may affect what features may be exercised by
+  WGSL programs. See [WebGPU] Section 3.6 Optional capabilities,
+  and [WGSL] Section 11. Language Extensions.
+
+  It is expected that implementations will behave as if this
+  registration applies to later editions of [WGSL], and its published
+  specification references may be updated accordingly from time to
+  time.  Although this expectation is unusual among media type
+  registrations, it matches widespread industry conventions.
+
+Published specification: [WGSL]
+
+Applications that use this media type:
+
+  Implementations of WebGPU [WebGPU]. This is expected to include
+  web browsers.
+
+Fragment identifier considerations: None
+
+Additional information:
+
+  Magic number(s): None
+  File extension(s): .wgsl
+  Macintosh file type code(s): TEXT
```

Comment:
For Macintosh file type, I copied what text/javascript has

I guess @litherum can advise